backend/src/zac/contrib/dowc/views.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any, NoReturn, Optional
+
+from django.http import HttpResponse
+
+from rest_framework import authentication, permissions, status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from zac.core.services import find_document
+
+from .api import get_doc_info, patch_and_destroy_doc
+from .permissions import CanOpenDocuments
+from .serializers import DowcResponseSerializer
+
+
+def _cast(value: Optional[Any], type_: type) -> Any:
+    if value is None:
+        return value
+    return type_(value)
+
+
+class OpenDowcView(APIView):
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated & CanOpenDocuments,)
+    document = None
+    serializer_class = DowcResponseSerializer
+
+    def get_object(self) -> NoReturn:
```

Comment:
```suggestion
    def get_object(self) -> Document:
```
with `zgw_consumers.api_models.documenten.Document`